sdk/src/contexts/abstracts/tokenBridge.ts

```diff
@@ -36,51 +73,118 @@ export abstract class TokenBridgeAbstract<TransactionResult> {
+   * @returns The token address as a 32-byte Wormhole address
+   */
+  protected abstract formatAssetAddress(address: string): Promise<any>;
+  /**
+   * Parse a token address from a 32-byte Wormhole address which is the standard format for a given chain
+   *
+   * @param address The 32-byte wormhole address
+   * @returns The token address in the blockchain specific format
+   */
+  protected abstract parseAssetAddress(address: any): Promise<string>;
 
+  /**
+   * Fetches the address for a token on any chain (These are the Wormhole token addresses, not necessarily the cannonical version of that token)
+   *
+   * @param tokenId The Token ID (chain/address)
+   * @param chain The chain name or id
+   * @returns The Wormhole address on the given chain, null if it does not exist
+   */
   protected abstract getForeignAsset(
     tokenId: TokenId,
     chain: ChainName | ChainId,
   ): Promise<string | null>;
+  /**
+   * Fetches the address for a token on any chain (These are the Wormhole token addresses, not necessarily the cannonical version of that token)
+   *
+   * @param tokenId The Token ID (chain/address)
+   * @param chain The chain name or id
+   * @returns The Wormhole address on the given chain
+   * @throws Throws if the token does not exist
+   */
   protected abstract mustGetForeignAsset(
     tokenId: TokenId,
     chain: ChainName | ChainId,
   ): Promise<string>;
+  /**
+   * Parses all relevant information from a transaction given the sending tx hash and sending chain
+   *
+   * @param tx The sending transaction hash
+   * @param chain The sending chain name or id
+   * @returns The parsed data
+   */
   protected abstract parseMessageFromTx(
     tx: string,
     chain: ChainName | ChainId,
   ): Promise<ParsedMessage[] | ParsedRelayerMessage[]>;
 
+  /**
+   * Fetches the native token balance for a wallet
+   *
+   * @param walletAddress The wallet address
+   * @param chain The chain name or id
+   * @returns The native balance as a BigNumber
+   */
   protected abstract getNativeBalance(
     walletAddress: string,
     chain: ChainName | ChainId,
   ): Promise<BigNumber>;
+  /**
+   * Fetches the balance of a given token for a wallet
+   *
+   * @param walletAddress The wallet address
+   * @param tokenId The token ID (its home chain and address on the home chain)
+   * @param chain The chain name or id
+   * @returns The token balance of the wormhole asset as a BigNumber
+   */
   protected abstract getTokenBalance(
     walletAddress: string,
     tokenId: TokenId,
     chain: ChainName | ChainId,
   ): Promise<BigNumber | null>;
 
+  /**
+   * Redeems funds for a token bridge transfer on the destination chain
+   *
+   * @param destChain The destination chain name or id
+   * @param signedVAA The Signed VAA bytes
+   * @param overrides Optional overrides, varies between chains
+   * @param payerAddr Optional, Solana specific. The address that pays for the redeem transaction
```

Comment:
The `payerAddr` might not be specific to Solana. We can just specify that it's the address that will send the transaction/pay the fees